src/SerializeAST.cc

```diff
@@ -773,3 +863,36 @@ sos::Object drafter::WrapBlueprint(const Blueprint& blueprint)
 
     return blueprintObject;
 }
+
+sos::Object drafter::WrapBlueprint(const Blueprint& blueprint, const ASTType astType, bool expand)
+{
+    sos::Object blueprintObject;
+
+#if _WITH_REFRACT_
+    registerNamedTypes(blueprint.content.elements());
+    ExpandMSON = expand;
```

Comment:
We still need this because expansion needs to happen for rendering. The ExpandMSON variable is for whether we need to expand MSON in the Refract AST.

--
	Pavan Kumar Sunkara
Blueprint Guy, apiary.io
pavan@apiary.io | pksunkara | @pksunkara

On 11 September 2015 at 5:08:03 pm, klokane (notifications@github.com) wrote:

In src/SerializeAST.cc:

> @@ -773,3 +863,36 @@ sos::Object drafter::WrapBlueprint(const Blueprint& blueprint)
>   
>      return blueprintObject;
>  }
> +
> +sos::Object drafter::WrapBlueprint(const Blueprint& blueprint, const ASTType astType, bool expand)
> +{
> +    sos::Object blueprintObject;
> +
> +#if _WITH_REFRACT_
> +    registerNamedTypes(blueprint.content.elements());
> +    ExpandMSON = expand;
we need no call registerNamedTypes() if no expansion happen

—
Reply to this email directly or view it on GitHub.